src/app/containers/Timestamp/index.jsx

```diff
@@ -1,50 +1,95 @@
-import React from 'react';
+import React, { Fragment } from 'react';
 import { number } from 'prop-types';
 import Timestamp from '../../components/Timestamp';
+import relativeTime from './relativeTimestamp';
 
 // if the date is invalid return null - https://stackoverflow.com/questions/1353684/detecting-an-invalid-date-date-instance-in-javascript#answer-1353711
 const isValidDateTime = dateTime => !isNaN(dateTime); // eslint-disable-line no-restricted-globals
 
-const formatDateTime = dateObj => {
-  const fullYear = dateObj.getFullYear();
-  const monthTwoDigit = dateObj.toLocaleDateString('en-GB', {
-    month: '2-digit',
-  });
-  const dayTwoDigit = dateObj.toLocaleDateString('en-GB', {
-    day: '2-digit',
-  });
+const longNumeric = {
+  locale: 'en-GB',
+  month: '2-digit',
+  day: '2-digit',
+  reverse: true,
+  separator: '-',
+};
 
-  return `${fullYear}-${monthTwoDigit}-${dayTwoDigit}`;
+const shortAlphaNumeric = {
+  locale: 'en-GB',
+  month: 'long',
+  day: 'numeric',
+  reverse: false,
+  separator: ' ',
 };
 
-const formatTimestamp = dateObj => {
+const formatUnixTimestamp = (milliseconds, formatType) => {
+  const dateObj = new Date(milliseconds);
   const fullYear = dateObj.getFullYear();
-  const monthLong = dateObj.toLocaleDateString('en-GB', {
-    month: 'long',
+  const month = dateObj.toLocaleDateString('en-GB', {
+    month: formatType.month,
   });
-  const dayNumeric = dateObj.toLocaleDateString('en-GB', {
-    day: 'numeric',
+  const day = dateObj.toLocaleDateString('en-GB', {
+    day: formatType.day,
   });
 
-  return `${dayNumeric} ${monthLong} ${fullYear}`;
+  const orderedDate = formatType.reverse
+    ? [day, month, fullYear].reverse()
+    : [day, month, fullYear];
+
+  return orderedDate.join(formatType.separator);
+};
+
+const isTenHoursAgoOrLess = milliseconds => {
+  const now = Date.now();
+  return now - milliseconds >= 10 * 60 * 60 * 1000;
 };
 
-const TimestampContainer = ({ timestamp }) => {
-  const dateObj = new Date(timestamp);
+const timestampWithPrefixUpdated = (datetime, updateTime) => (
+  <Timestamp datetime={datetime} prefix="Updated ">
+    {updateTime}
+  </Timestamp>
+);
+
+const defaultTimestamp = published => (
+  <Timestamp datetime={formatUnixTimestamp(new Date(published), longNumeric)}>
+    {formatUnixTimestamp(new Date(published), shortAlphaNumeric)}
+  </Timestamp>
+);
+
+const hasBeenUpdated = (updated, published) => updated !== published;
+
+const updatedTimestamp = (updated, published) => {
+  if (!hasBeenUpdated(updated, published)) {
+    return null;
+  }
+
+  return timestampWithPrefixUpdated(
+    formatUnixTimestamp(updated, longNumeric),
+    isTenHoursAgoOrLess(updated)
+      ? formatUnixTimestamp(updated, shortAlphaNumeric)
+      : relativeTime(updated),
+  );
```

Comment:
Quite a lot going on in this function, tricky to mentally parse. Not sure how it could be improved... 🤔 maybe a comment or two?

Reply:
https://github.com/bbc/simorgh/blob/10e98f946cf722b82c29beae2270adfcc1c5ad62/src/app/containers/Timestamp/index.jsx#L66 added a comment here.. is that okay?